### ISSUE TITLE
Remove the Editor Iframe

### ISSUE THREAD

**Issue by Copons** [Automattic/wp-calypso#95393]
Some useful resources:

- From the olden Cylon days:
  - p58i-eb7-p2
  - p4TIVU-atS-p2
- pfsHM7-4W-p2 - latest discussion around this (contains backlinks to previous discussions)
- p1717771015925179/1717766189.625569-slack-C029GN3KD
- https://github.com/Automattic/wp-calypso/pull/87120 - PR that was fully working before we decided to abandon it due to the blockers mentioned in the post.

### Update 2025-01-16

Every time we untangle a screen linking to the editor, we remove an entry point to the editor iframe.

For example, by Untangling the Posts list, we "removed" the editor iframe for the Post post type.

With this approach, the editor iframe will go away "automatically", and we'll be only left with cleaning up the old code.

**Comment by fushar** [comment]
Adding useful resources:

- pfsHM7-4W-p2 - latest discussion around this (contains backlinks to previous discussions)
- https://github.com/Automattic/wp-calypso/pull/87120 - PR that was fully working before we decided to abandon it due to the blockers mentioned in the post.

**Comment by fushar** [comment]
Adding a comparison of some features (that will go away) in the iframed editor, vs Core:

|Feature|iframed editor|Core|
|-|-|-|
|Media Library|![Image](https://github.com/user-attachments/assets/e297957d-62d0-4c15-90f4-d20cc86ecfc5)|![Image](https://github.com/user-attachments/assets/fef75ad4-497f-4d40-ba37-53920a640219)|
|Revisions|![Image](https://github.com/user-attachments/assets/99067fe4-60bd-4ca8-aa9b-ca4cf7e5a24a)|![Image](https://github.com/user-attachments/assets/28e94a16-012c-465c-b5fe-c1a5027f441a)|

**Comment by Copons** [comment]
Good point @fushar! Those are actual blockers for this issue, and we need to wait for Core editor updates to fill the gaps.

@richtabor I think revisions and media library were briefly mentioned in the check-in. Do you have any specific details about them? Otherwise, we should discuss them at the next check-in.

**Comment by richtabor** [comment]
Of these here, can you list for each what we'd want to consider bringing to core? 

What exactly are the changes for each that Calypso does better, and do customers rely on those changes today?

**Comment by Copons** [comment]
I was under the impression that we were already working to bring some of those features to Core, but I must have misunderstood. So, a few things off the top of my head:

### Revisions

I'd argue that neither Core nor Calypso does a particularly good job, especially with blocks and the visual clutter they bring in code view. However...

Calypso has:
* **Toggle between unified and split views.**
* Edited lines diff.
* Visual list of revisions, as opposed to the slider.

Core has:
* Compare any two revisions.

### Media Library

Calypso has:
* **External sources.**
* Slider to change thumbnail sizes.
* Filter by file type or by files attached to the current post.

Core has:
* Filter by date.
* Edit file attributes in the sidebar, instead of navigating into its individual screen.
* Edit image (crop, rotate, etc.).
  * I recall this feature existing in Calypso as well, but I could not do it anymore. Not sure if it's a bug.

---

I'd say there are no winners overall. I've highlighted the Calypso features that I consider good ones for Core. Possibly, external sources in the media library is the only dealbreaker, but otherwise, I don't see any blockers. 🤔

**Comment by Copons** [comment]
As an update, we have agreed that Revisions are not a blocker, but we should investigate the Media Library integration with the editor, which might not have been looked at in years.

We'll do a spike ASAP to confirm whether it needs the iframe, or it could be integrated natively.

If that's the case, we won't have any significant blockers to **remove the iframe for all users**.

**Comment by simison** [comment]
> we should investigate the Media Library integration with the editor, which might not have been looked at in years.

@tyxla might have the most recent knowledge (internal ref p4TIVU-atS-p2 )

**Comment by tanjoymor** [comment]
From a Happiness / user perspective

Revisions: 
* Agree no blockers - users struggle with either version because its all code
* Atomic sites (even set to Default UI) already use wp-admin revision screen

Media Library:
* Media Library integration with the editor - currently on Simple and Atomic, the insert modal in the editor matches the Media Library view for the assigned UI - it might be weird to remove the iframe before the Calypso Media Library view is removed, but done at the same it, it would make sense.

The important editor piece are the Block options:
![Image](https://github.com/user-attachments/assets/43080ecd-6ef3-4d84-92ec-e9ae86485a1d)

**Comment by dsas** [comment]
> Calypso has:
>   External sources.

Google Photos / Pexels / etc all work in the un-iframed post editor already though right? I think Core provided an API to do it, at least for the media sidebar.

**Comment by tyxla** [comment]
> > we should investigate the Media Library integration with the editor, which might not have been looked at in years.
> 
> [@tyxla](https://github.com/tyxla) might have the most recent knowledge (internal ref p4TIVU-atS-p2 )

Unfortunately, I don't have much more context - we haven't done much in the past year other than some data store migrations last year. 

At this point, I believe it will be easier to list the features we want to keep from the Calypso media library and migrate them to using the relevant extension points in Gutenberg. This makes even more sense now that the site editor uses the core media library experience while the post editor uses the Calypso one, which is yet another point of confusion for WP.com users.

**Comment by simison** [comment]
If I remember right, in our past reviews for removing the iframe regarding media library:
- **UX that people like overall.** It is debatable if this should block from just removing the iframe; it's already gone in the site editor and for folks using Firefox and Safari when 3rd party cookies aren't enabled.
- **Media sources (Google, Pexels, Openverse)**. These were migrated directly to blocks so long ago that shouldn't be a reason to keep iframe anymore. Sure, it would be great to add also to the new media sidebar but that's a minor project that you should just get done regardless of iframe.
- **Storage upsells** I believe these were moved to wp-admin already, with iframe removal in mind, but I'm not entirely sure. I might've also just seen designs, but the project never shipped.

**Comment by fushar** [comment]
Sorry for jumping in a bit late, but what about Jetpack app upsell banners? They were pointed out as blocker here: pfsHM7-4W-p2#comment-172 (though we hadn't done any conversion analyses yet)

<img src="https://github.com/user-attachments/assets/64e77015-9ce1-44ab-9714-822a9be5efb1" width="300">

**Comment by Copons** [comment]
> what about Jetpack app upsell banners

@fushar `AppBanner` is a fairly simple component. I don't think there are any big challenges in moving it — at least the part pertaining to the "Rich mobile publishing" editor nudge — to the JP monorepo. 🤔 

It's also fairly ancient. I bet that rewritten in modern React would be 10x smaller!

**Comment by fushar** [comment]
Skipping the iframe is the easiest part; the harder one is to review / migrate the upsells if necessary (as I mentioned above).

**Comment by Copons** [comment]
### Update

Every time we untangle a screen linking to the editor, we remove an entry point to the editor iframe.

For example, by Untangling the Posts list, we "removed" the editor iframe for the Post post type.

With this approach, the editor iframe will go away "automatically", and we'll be only left with cleaning up the old code.

**Comment by tyxla** [comment]
@xavier-lc perhaps we can close this one now that the work to remove the iframed editor is finished?
